snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/rf2/SnomedRf2ExportRequest.java

```diff
@@ -15,273 +15,777 @@
  * @since 5.7
  */
-final class SnomedRf2ExportRequest implements Request<BranchContext, UUID> {
+final class SnomedRf2ExportRequest implements Request<RepositoryContext, UUID> {
 
 	private static final long serialVersionUID = 1L;
-	
+
+	private static final Pattern NAMESPACE_PATTERN = Pattern.compile("\\{(\\d{7})\\}");
+
+	private static final Ordering<CodeSystemVersionEntry> EFFECTIVE_DATE_ORDERING = Ordering.natural()
+			.onResultOf(CodeSystemVersionEntry::getEffectiveDate);
+
 	@JsonProperty
 	@NotEmpty
 	private String userId;
-	
+
 	@JsonProperty
 	@NotEmpty
 	private String codeSystem;
-	
+
 	@JsonProperty
-	private boolean includeUnpublished;
-	
+	@NotEmpty
+	private String referenceBranch;
+
 	@JsonProperty 
 	@NotNull 
 	private Rf2ReleaseType releaseType;
-	
-	@JsonProperty 
-	private boolean extensionOnly;
-	
-	@JsonProperty 
-	private String startEffectiveTime;
-	
+
+	@JsonProperty
+	@NotNull
+	private Rf2MaintainerType maintainerType;
+
+	@JsonProperty
+	@NotNull
+	private Rf2RefSetExportLayout refSetExportLayout;
+
+	@JsonProperty
+	private String nrcCountryCode;
+
+	@JsonProperty
+	private String namespaceId;
+
 	@JsonProperty 
-	private String endEffectiveTime;
-	
+	private Date startEffectiveTime;
+
 	@JsonProperty 
-	private Collection<String> modules;
-	
+	private Date endEffectiveTime;
+
 	@JsonProperty
-	private String transientEffectiveTime;
-	
+	private boolean includePreReleaseContent;
+
 	@JsonProperty
-	private String namespace;
-	
+	private Collection<String> componentTypes;
+
+	@JsonProperty 
+	private Collection<String> modules;
+
 	@JsonProperty
 	private Collection<String> refSets;
-	
+
 	@JsonProperty
-	private Collection<String> componentTypes;
+	private String transientEffectiveTime;
+
+	@JsonProperty 
+	private boolean extensionOnly;
 
 	SnomedRf2ExportRequest() {}
-	
-	void setUserId(String userId) {
+
+	void setUserId(final String userId) {
 		this.userId = userId;
 	}
-	
-	void setCodeSystem(String codeSystem) {
+
+	void setCodeSystem(final String codeSystem) {
 		this.codeSystem = codeSystem;
 	}
 
-	void setIncludeUnpublished(boolean includeUnpublished) {
-		this.includeUnpublished = includeUnpublished;
+	void setReferenceBranch(final String referenceBranch) {
+		this.referenceBranch = referenceBranch;
 	}
-	
-	void setReleaseType(Rf2ReleaseType releaseType) {
+
+	void setReleaseType(final Rf2ReleaseType releaseType) {
 		this.releaseType = releaseType;
 	}
-	
-	void setExtensionOnly(boolean extensionOnly) {
-		this.extensionOnly = extensionOnly;
+
+	void setMaintainerType(final Rf2MaintainerType maintainerType) {
+		this.maintainerType = maintainerType;
 	}
-	
-	void setStartEffectiveTime(String startEffectiveTime) {
+
+	void setRefSetExportLayout(final Rf2RefSetExportLayout refSetExportLayout) {
+		this.refSetExportLayout = refSetExportLayout;		
+	}
+
+	void setNrcCountryCode(final String nrcCountryCode) {
+		this.nrcCountryCode = nrcCountryCode;
+	}
+
+	void setNamespaceId(final String namespaceId) {
```

Comment:
Is this the ID of the namespace concept? Wouldn't it be better to use the namespace ID directly? LanguageSetting is going to be deprecated soon, so we either check all FSNs or outsource namespace ID extraction to the clients.

Reply:
Moving computation to the client.